src/pydata_sphinx_theme/assets/styles/sections/_header.scss

```diff
@@ -107,10 +107,26 @@
         border: 1px solid var(--pst-color-border);
         box-shadow: 0 0 0.3rem 0.1rem var(--pst-color-shadow);
         background-color: var(--pst-color-on-background);
-        padding: 0.5rem 1rem;
+        padding: 0.5rem 0;
         margin: 0.5rem 0;
         min-width: 20rem;
 
+        .dropdown-item {
+          // Give the items in the dropdown some breathing room but let the hit
+          // and hover area of the items extend to the edges of the menu
+          padding: 0.25rem 1.5rem;
+
+          // Override Bootstrap
+          &:focus:not(:hover):not(:active) {
```

Comment:
Had to add the `:not(:hover):not(:active)`, otherwise this focus rule overrides the Bootstrap-provided hover and active styles because the selector has higher specificity.